sound/soc/sof/topology.c

```diff
@@ -1743,6 +1750,12 @@ static int sof_dai_load(struct snd_soc_component *scomp, int index,
 		goto free_playback_tables;
 	}
 
+	if (ipc_tplg_ops->dai_load) {
+		ret = ipc_tplg_ops->dai_load(sdev, spcm, stream);
+		if (ret < 0)
+			snd_dma_free_pages(&spcm->stream[stream].page_table);
```

Comment:
if there is capture and playback, don't we need to free the playback DMA pages as well? I there a missing goto free_playback_tables here?

And if the dai_load fails here, don't we need a dai_unload on the playback?

Looks like the error flow needs more work.

Reply:
sure, good catch. thanks!